src/pilot/assign_role.py

```diff
@@ -881,13 +881,15 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
     controllers = drac_client.list_raid_controllers()
 
     found_hba = False
+    found_boss = False
     for controller in controllers:
         if "hba330" in controller.model.lower():
             found_hba = True
-            break
+        if "boss" in controller.model.lower():
+            found_boss = True
 
-    if not found_hba:
-        LOG.info("No HBA330 found.  Not generating OSD config for "
+    if not (found_hba and found_boss):
+        LOG.info("Both BOSS and HBA330 are not present.  Not generating OSD config for "
```

Comment:
Log message wording is awkward.  Suggest "Neither BOSS nor HBA330 are present" to make more clear.

Reply:
agreed

Reply:
The intent is that BOTH the boss card and the hba must be present for us to generate the OSD config.  Your wording makes it sound like we would generate the config if either one is present.

Reply:
Ok, my mistake. Agreed with your original wording then!